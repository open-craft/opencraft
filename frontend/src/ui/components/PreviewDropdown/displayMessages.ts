const messages = {
  dashboard: {
    defaultMessage: 'Dashboard',
    description: 'Dashboard dropdown item'
  },
  courseware: {
    defaultMessage: 'Courseware',
    description: 'Courseware dropdown item'
  }
};

export default messages;
