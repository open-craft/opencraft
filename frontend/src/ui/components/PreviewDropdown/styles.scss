@import '~styles/theme';

div.dropdown-container {
    display: flex;
    flex-direction: column;
    align-items:center;
    position: relative;
    width: 25%;
    height: 100%;
    z-index: 1;

    a {
        width: 100%;
        padding-left: 0px;
    }

    .fa-chevron-up, .fa-chevron-down {
        margin-left: 5px;
    }

    .dropdown-items {
        width: 100%;
        align-items: center;
        box-shadow: 3px 5px 5px $secondary-2;
        background-color: white;

        div.content {
            position:relative;
            z-index: 0;
            width: 100%;
            display: flex;
            flex-direction: column;
            font-size: 0;

            button {
                display:inline-block;
                color: $primary-1;
                font-size: 14px;
                font-family : $chivo-font;
                margin: 0px;
                height: 44px;
                width: 100%;
                line-height: 20px;
                background-color:white;
                border-width: 0px;
            }
            button:hover{
                background-color: $light-blue-2;
            }

            p {
                margin-top:10px;
            }
        }
    }

}
