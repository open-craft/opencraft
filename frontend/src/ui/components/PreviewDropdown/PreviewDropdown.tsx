import * as React from 'react';
import { Collapse, Nav } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import messages from './displayMessages';
import './styles.scss';

interface PreviewDropdownProps {
  handleChange: Function;
}

export const PreviewDropdown: React.FC<PreviewDropdownProps> = (
  props: PreviewDropdownProps
) => {
  const [open, setOpen] = React.useState(false);
  const [selectedPreview, setSelectedPreview] = React.useState('dashboard');
  const updatePreview = (key: string) => {
    setSelectedPreview(key);
    setOpen(!open);
    props.handleChange(key);
  };
  const iconName = open ? 'fa-chevron-up' : 'fa-chevron-down';
  const children = () => {
    const toRender: JSX.Element[] = [];

    Object.entries(messages).forEach(([key, _]) =>
      toRender.push(
        <button
          onClick={() => updatePreview(key)}
          key={key}
          type="button"
          className="d-inline-block"
        >
          <p>
            <WrappedMessage messages={messages} id={key} />
          </p>
        </button>
      )
    );

    return <div>{toRender}</div>;
  };

  return (
    <div className="dropdown-container d-flex flex-column align-items-center">
      <Nav.Link onClick={() => setOpen(!open)}>
        <WrappedMessage messages={messages} id={selectedPreview} />
        <i className={`fas xs ${iconName}`} />
      </Nav.Link>
      <span className="fill-line" />
      <div className="dropdown-items align-items-center">
        <div className="content d-flex flex-column">
          <Collapse in={open}>{children()}</Collapse>
        </div>
      </div>
    </div>
  );
};
