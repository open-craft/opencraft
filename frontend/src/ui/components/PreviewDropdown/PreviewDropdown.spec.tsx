import React from 'react';
import {
    mountComponentForTesting,
    setupComponentForTesting
} from "utils/testing";
import { PreviewDropdown } from './PreviewDropdown';
import messages from './displayMessages'

describe("Preview dropdown", function() {
    it('renders without crashing', () => {
        const tree = setupComponentForTesting(<PreviewDropdown />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders messages as options', () => {
        messages['somepage'] = {defaultMessage: 'A new page', description: ''};
        const tree = mountComponentForTesting(<PreviewDropdown />);
        const mountedComponent = tree.find('.dropdown-items').find('FormattedMessage');
        expect(mountedComponent.length).toBe(3)
        expect(mountedComponent.children().debug()).toBe('Dashboard\n\n\nCourse Outline\n\n\nA new page')
        expect(tree).toMatchSnapshot();
    });
});
