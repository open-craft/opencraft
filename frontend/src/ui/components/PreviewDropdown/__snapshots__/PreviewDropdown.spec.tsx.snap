// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview dropdown renders messages as options 1`] = `ReactWrapper {}`;

exports[`Preview dropdown renders without crashing 1`] = `
<div
  className="dropdown-container"
>
  <a
    className="nav-link"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    Dashboard
    <i
      className="fas xs fa-chevron-down"
    />
  </a>
  <span
    className="fill-line"
  />
  <div
    className="dropdown-items"
  >
    <div
      className="content"
    >
      <div
        aria-expanded={null}
        className="collapse"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          <p>
            Dashboard
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
